conformance_tests/test_tiovx/test_bidir_ext_standardnode.c

```diff
@@ -0,0 +1,213 @@
+#include <TI/tivx.h>
+#include <VX/vx_khr_bidirectional_parameters.h>
+
+TESTCASE(bpExtStandardNodes, CT_VXContext, ct_setup_vx_context, 0)
+
+/* Write some data on an 8-bit image */
+void bd_std_node_writeImage(vx_image image, vx_uint8 a, vx_uint8 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa) = a;
+    *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa) = b;
+    vxUnmapImagePatch(image, id);
+}
+
+/* Check data in an 8-bit image */
+vx_status bd_std_node_checkImage(vx_image image, vx_uint8 a, vx_uint8 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    vx_uint8 aa = *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa);
+    vx_uint8 bb = *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa);
+    vxUnmapImagePatch(image, id);
+    /* On failure return a non-zero value encoding the pixels, easily readable in decimal */
+    return (aa == a) && (bb  == b) ? VX_SUCCESS : aa * 1000 + bb + 1000000;
+}
+
+/* Write some data on a 16-bit image */
+void bd_std_node_writeImageS16(vx_image image, vx_int16 a, vx_int16 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa) = a;
+    *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa) = b;
+    vxUnmapImagePatch(image, id);
+}
+
+/* Check data in an 8-bit image */
+vx_status checkImageS16(vx_image image, vx_int16 a, vx_int16 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    vx_int16 aa = *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa);
+    vx_int16 bb = *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa);
+    vxUnmapImagePatch(image, id);
+    /* On failure output the data and return VX_FAILURE */
+    if (aa == a && bb == b)
+        return VX_SUCCESS;
+    printf("got %d and %d when %d and %d was expected\n", (int)aa, (int)bb, (int)a, (int)b);
+    return VX_FAILURE;
+}
+
+vx_status releaseNode(vx_node node)
+{
+    return vxReleaseNode(&node);
+}
+
+void testAccumulate(vx_context context)
+{
+    vx_graph graph = vxCreateGraph(context);
+    vx_image in_param = vxCreateImage(context, 8, 8, VX_DF_IMAGE_U8);
```

Comment:
Should we add some checks to confirm no errors creating these objects?

Reply:
look how it is done in the TI test framework for the vxCreateGraph and vxCreateImage. @RB